src/REPLICA/fix_pimd_langevin.cpp

```diff
@@ -0,0 +1,1497 @@
+using MathConst::MY_PI;
+using MathConst::THIRD;
+
+enum { NMPIMD };
+enum { PHYSICAL, NORMAL };
+enum { BAOAB, OBABO };
+enum { ISO, ANISO, TRICLINIC };
+enum { PILE_L };
+enum { MTTK, BZP };
+
+static std::map<int, std::string> Barostats{{MTTK, "MTTK"}, {BZP, "BZP"}};
+enum { NVE, NVT, NPH, NPT };
+enum { SINGLE_PROC, MULTI_PROC };
+
+/* ---------------------------------------------------------------------- */
+
+FixPIMDLangevin::FixPIMDLangevin(LAMMPS *lmp, int narg, char **arg) :
+    Fix(lmp, narg, arg), random(nullptr), c_pe(nullptr), c_press(nullptr)
+{
+  restart_global = 1;
+  time_integrate = 1;
+  tagsend = tagrecv = nullptr;
+  bufsend = bufrecv = nullptr;
+  bufsendall = bufrecvall = nullptr;
+  bufsorted = bufsortedall = nullptr;
+  outsorted = buftransall = nullptr;
+
+  ntotal = 0;
+  maxlocal = maxunwrap = maxxc = 0;
+  bufbeads = nullptr;
+  x_unwrap = xc = nullptr;
+  xcall = nullptr;
+  counts = nullptr;
+
+  sizeplan = 0;
+  plansend = planrecv = nullptr;
+
+  M_x2xp = M_xp2x = M_f2fp = M_fp2f = nullptr;
+  lam = nullptr;
+  modeindex = nullptr;
+
+  mass = nullptr;
+
+  method = NMPIMD;
+  ensemble = NVT;
+  integrator = OBABO;
+  thermostat = PILE_L;
+  barostat = BZP;
+  fmass = 1.0;
+  np = universe->nworlds;
+  sp = 1.0;
+  temp = 298.15;
+  Lan_temp = 298.15;
+  tau = 1.0;
+  tau_p = 1.0;
+  Pext = 1.0;
+  pdim = 0;
+  pilescale = 1.0;
+  tstat_flag = 1;
+  pstat_flag = 0;
+  mapflag = 1;
+  removecomflag = 1;
+  fmmode = PHYSICAL;
+  pstyle = ISO;
+  totenthalpy = 0.0;
+
+  int seed = -1;
+
+  for (int i = 0; i < 6; i++) {
+    p_flag[i] = 0;
+    p_target[i] = 0.0;
+  }
+
+  for (int i = 3; i < narg - 1; i += 2) {
+    if (strcmp(arg[i], "method") == 0) {
+      if (strcmp(arg[i + 1], "nmpimd") == 0)
+        method = NMPIMD;
+      else
+        error->universe_all(FLERR, "Unknown method parameter for fix pimd/langevin");
+    } else if (strcmp(arg[i], "integrator") == 0) {
+      if (strcmp(arg[i + 1], "obabo") == 0)
+        integrator = OBABO;
+      else if (strcmp(arg[i + 1], "baoab") == 0)
+        integrator = BAOAB;
+      else
+        error->universe_all(
+            FLERR,
+            "Unknown integrator parameter for fix pimd/langevin. Only obabo and baoab "
+            "integrators are supported!");
+    } else if (strcmp(arg[i], "ensemble") == 0) {
+      if (strcmp(arg[i + 1], "nve") == 0) {
+        ensemble = NVE;
+        tstat_flag = 0;
+        pstat_flag = 0;
+      } else if (strcmp(arg[i + 1], "nvt") == 0) {
+        ensemble = NVT;
+        tstat_flag = 1;
+        pstat_flag = 0;
+      } else if (strcmp(arg[i + 1], "nph") == 0) {
+        ensemble = NPH;
+        tstat_flag = 0;
+        pstat_flag = 1;
+      } else if (strcmp(arg[i + 1], "npt") == 0) {
+        ensemble = NPT;
+        tstat_flag = 1;
+        pstat_flag = 1;
+      } else
+        error->universe_all(FLERR,
+                            "Unknown ensemble parameter for fix pimd/langevin. Only nve and nvt "
```

Comment:
Apparently "nph" and "npt" are supported now, too. So the error message should be updated.

Reply:
solved by [f838677](https://github.com/lammps/lammps/pull/3660/commits/f83867787c6eb7ed8b509a54be25b3ee46e7290d)